### PR TITLE
[webkitpy] Use canonical-id in suggestnominations

### PR DESCRIPTION
**Pull request by dpino**
#### bae60fe2c5a7a0942f0440a004d6ec489e42f963
<pre>
[webkitpy] Use canonical-id in suggestnominations
<a href="https://bugs.webkit.org/show_bug.cgi?id=245277">https://bugs.webkit.org/show_bug.cgi?id=245277</a>

Reviewed by Jonathan Bedard.

Since WebKit moved to GitHub, commit messages no longer include the
&apos;svn-id&apos; trailer, which was used to compile statistics about number of
commits of potential commiter and reviewer candidates.  This identifier
should now be replaced by &apos;canonical-id&apos;.

The committer message regular expression had to be modified as it was
failing many times and not all commit messages were processed. Also do
not raise exception if committer could not be found contributors list.

* Tools/Scripts/webkitpy/tool/commands/suggestnominations.py:
(AbstractCommitLogCommand):
(AbstractCommitLogCommand._parse_commit_message):
(SuggestNominations._count_commit):
* Tools/Scripts/webkitpy/tool/commands/suggestnominations_unittest.py:
(SuggestNominationsTest):

Canonical link: <a href="https://commits.webkit.org/254701@main">https://commits.webkit.org/254701@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/fec75e4e0ba1b5ab43588eb5314ca288da2f7237

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/89468 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/34022 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/20208 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/98788 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/155094 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/93476 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/32527 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/28014 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/81828 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/93200 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/95115 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/68/builds/25840 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/76366 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/25768 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/93053 "Passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/80718 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/80616 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/68757 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/30286 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/14690 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/30024 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/15627 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3302 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/33471 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/38584 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/32181 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/34718 "Passed tests") | |

### CODE REVIEW

**Comment by dpino** [comment]
Results with patch applied:

```bash
$ ./Tools/Scripts/webkit-patch suggest-nominations
COMMITTER: Ahmad Saleem (ahmad.saleem792@gmail.com) has 12 reviewed patches
COMMITTER: Alexey Knyazev (3479527+lexaknyazev@users.noreply.github.com) has 35 reviewed patches
COMMITTER: Asumu Takikawa (asumu@igalia.com) has 17 reviewed patches
COMMITTER: Commit Queue (commit-queue@webkit.org) has 11 reviewed patches
COMMITTER: Geza Lore (glore@igalia.com) has 17 reviewed patches
COMMITTER: James Hilliard (james.hilliard1@gmail.com) has 18 reviewed patches
COMMITTER: Joseph Griego (jgriego@igalia.com) has 12 reviewed patches
COMMITTER: Olivier Blin (olivier.blin@softathome.com) has 10 reviewed patches
COMMITTER: Richard Robinson (richard_robinson2@apple.com) has 10 reviewed patches
COMMITTER: Rupin Mittal (rmittal2@apple.com) has 12 reviewed patches
COMMITTER: Ryan Reno (rreno@apple.com) has 10 reviewed patches
REVIEWER: Tyler Wilcock (tyler_w@apple.com) has 132 reviewed patches
COMMITTER: Vitor Roriz (vitor.roriz@apple.com) has 10 reviewed patches
COMMITTER: eocanha@igalia.com (eocanha@igalia.com@268f45cc-cd09-0410-ab3c-d52691b4dbfc) has 11 reviewed patches
COMMITTER: zan@falconsigh.net (zan@falconsigh.net@268f45cc-cd09-0410-ab3c-d52691b4dbfc) has 25 reviewed patches
```

File: Tools/Scripts/webkitpy/tool/commands/suggestnominations.py [hunk 1]
```diff
@@ -96,18 +96,16 @@ def _parse_commit_message(self, commit_message):
...
         commit_date_match = self._date_regexp.search(commit_message)
         if not commit_date_match:
             raise CommitLogError
         commit_date = commit_date_match.group('date')
 
-        revision_match = self._revision_regexp.search(commit_message)
-        if not revision_match:
+        canonical_match = self._canonical_regexp.search(commit_message)
+        if not canonical_match:
             raise CommitLogError
```
Comment (JonWBedard) [inline Tools/Scripts/webkitpy/tool/commands/suggestnominations.py]: I need to dig through exactly the consequences of this, but there are 2 commits in our history which don't have canonical links.

Comment (dpino) [reply Tools/Scripts/webkitpy/tool/commands/suggestnominations.py]: Any thoughts about this? AFAICT this change only affects the `suggestnomiations` command. 

It seems to me that for those commits that don't have a canonical id assigned there won't be `canonical_match` and the function will exit. In other words, those commits are not considered for computing potential committer or reviewers.

Comment (JonWBedard) [reply Tools/Scripts/webkitpy/tool/commands/suggestnominations.py]: Didn't get a chance to read through the surrounding code last week, your assessment is totally correct. This is exactly the sort of "consequence" that's perfectly acceptable in this function.

**Review by JonWBedard — APPROVED** [review]

**Pull request merged**